tartufo/scanner.py

```diff
@@ -683,3 +685,51 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path, {})
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    target: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        target: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param target: The local filesystem path to scan
+        """
+        self.target = target
+        super().__init__(global_options)
+
+    @property
+    def chunks(self) -> Generator[types.Chunk, None, None]:
+        """Yield the individual files in the target directory.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            yield types.Chunk(blob, file_path, {})
+
+    def _iter_folder(self) -> Generator[Tuple[str, str], None, None]:
+        folder_path = pathlib.Path(self.target)
+        for file_path in folder_path.rglob("**/*"):
+            relative_path = file_path.relative_to(folder_path)
+            if file_path.is_file() and self.should_scan(str(relative_path)):
+                try:
+                    with file_path.open("rb") as fhd:
+                        data = fhd.read()
+                except OSError as exc:
+                    raise click.FileError(filename=str(file_path), hint=str(exc))
+
+                try:
+                    blob = data.decode("utf-8")
+                except UnicodeDecodeError:
+                    # binary file, continue
+                    continue
```

Comment:
I'm not meaning to pick on this scanner in particular, but I'm not sure this is the proper approach. If the target file happens to be (I tried this as a test) a SQLite database that might contain tons of sensitive data, consider:

```python
>>> with open("gerbera.db", "rb") as f:
>>>     x = f.read()
>>> y = x.decode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa8 in position 35: invalid start byte
>>> y = str(x)
>>> "Adagio" in x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
>>> "Adagio" in y
True
```
Abstractly, I'd be happier if we tried to just look at the bytes (which might or might not be useful -- after all, maybe the input file is in EBCDIC), since in many cases we might in fact be able to see interesting content. The output of a match might be really horrific, so maybe this should just be food for future thought.

Reply:
Ya. Sounds like a nice feature that should be applied overall. For now, I think we just scan non-binary files.